# Report for ForestQuery into Global Deforestation, 1990 to 2016

ForestQuery is on a mission to combat deforestation around the world and to raise awareness about this topic and its impact on the environment. The data analysis team at ForestQuery has obtained data from the World Bank that includes forest area and total land area by country and year from 1990 to 2016, as well as a table of countries and the regions to which they belong.

The data analysis team has used SQL to bring these tables together and to query them in an effort to find areas of concern as well as areas that present an opportunity to learn from successes.


In [1]:
print("Executing postgresql setup...")
%reload_ext sql
%sql postgresql://jadug.parusa:@localhost/deforestation
%sql SELECT * FROM forest_area WHERE forest_area_sqkm>0 ORDER BY 3 DESC, 4 DESC LIMIT 3

Executing postgresql setup...
 * postgresql://jadug.parusa:***@localhost/deforestation
3 rows affected.


[('WLD', 'World', 2016, Decimal('39958245.9')),
 ('RUS', 'Russian Federation', 2016, Decimal('8148895')),
 ('BRA', 'Brazil', 2016, Decimal('4925540'))]

In [2]:
%%sql

DROP VIEW IF EXISTS deforestation_tab;
CREATE VIEW deforestation_tab AS (
    SELECT
        f.country_code,
        f.country_name,
        f.year,
        f.forest_area_sqkm,
        (2.59 * l.total_area_sqmi) AS total_area_sqkm,
        r.region,
        r.income_group
    FROM forest_area AS f
    JOIN
        land_area AS l
        ON
            f.country_code = l.country_code
        AND
            f.year = l.year
    JOIN
        regions AS r
        ON
            f.country_code = r.country_code
);
SELECT *, COUNT(*) over()
FROM deforestation_tab
WHERE forest_area_sqkm>0
GROUP BY 1,2,3,4,5,6,7
ORDER BY 3 DESC, 4 DESC
LIMIT 3;

 * postgresql://jadug.parusa:***@localhost/deforestation
Done.
Done.
3 rows affected.


[('WLD', 'World', 2016, Decimal('39958245.9'), Decimal('49171676.23'), 'World', 'NULL', 5570),
 ('RUS', 'Russian Federation', 2016, Decimal('8148895'), Decimal('6323115.83'), 'Europe & Central Asia', 'Upper middle income', 5570),
 ('BRA', 'Brazil', 2016, Decimal('4925540'), Decimal('3227081.08'), 'Latin America & Caribbean', 'Upper middle income', 5570)]

## 1. GLOBAL SITUATION

Question 1.1: How was the WORLD's 1990 total forest area compared to 2016?

In [21]:
%%sql
-- # calculate area, difference, and percentage
WITH
tab_1990_2016 AS (
    SELECT
        country_name,
        year,
        forest_area_sqkm
    FROM
        deforestation_tab
    WHERE
        country_name = 'World'
        AND (
            year = 1990
            OR year = 2016
        )
),
forest_1990 AS (
    SELECT
        forest_area_sqkm AS sqkm_base_1990
    FROM
        tab_1990_2016
    WHERE
        year = 1990
)

SELECT
    country_name,
    year,
    forest_area_sqkm,
    sqkm_base_1990,
    (forest_area_sqkm - sqkm_base_1990) AS loss_sqkm,
    100*ROUND((forest_area_sqkm - sqkm_base_1990) / sqkm_base_1990, 3) AS loss_percent
FROM
    tab_1990_2016 AS t,
    forest_1990 AS f
ORDER BY 2

 * postgresql://jadug.parusa:***@localhost/deforestation
2 rows affected.


[('World', 1990, Decimal('41282694.9'), Decimal('41282694.9'), Decimal('0.0'), Decimal('0.000')),
 ('World', 2016, Decimal('39958245.9'), Decimal('41282694.9'), Decimal('-1324449.0'), Decimal('-3.200'))]

In [25]:
%%sql
-- # filter 2016, order by total_area_sqmi
SELECT
    *
FROM
    deforestation_tab
WHERE
    year = 2016
    AND 
LIMIT 2

 * postgresql://jadug.parusa:***@localhost/deforestation
2 rows affected.


[('AFG', 'Afghanistan', 2016, Decimal('13500'), Decimal('252069.5'), 'South Asia', 'Low income'),
 ('ALB', 'Albania', 2016, Decimal('7705.39978'), Decimal('10579.15'), 'Europe & Central Asia', 'Upper middle income')]

According to the World Bank, the total forest area of the world was `41,282,694.9 sqkm` in 1990. As of 2016, the most recent year for which data was available, that number had fallen to `39,958,245.9 sqkm`, a loss of `-1,324,449.0 sqkm`, or `-3.200`%.

The forest area lost over this time period is slightly more than the entire land area of __________________ listed for the year 2016 (which is __________________).


## 2. REGIONAL OUTLOOK

In 2016, the percent of the total land area of the world designated as forest was __________________. The region with the highest relative forestation was__________________, with __________________%, and the region with the lowest relative forestation was __________________, with __________________% forestation.

In 1990, the percent of the total land area of the world designated as forest was __________________. The region with the highest relative forestation was__________________, with __________________%, and the region with the lowest relative forestation was __________________, with __________________% forestation.


The only regions of the world that decreased in percent forest area from 1990 to 2016 were __________________ (dropped from __________________% to __________________%) and __________________ (__________________% to __________________%). All other regions actually increased in forest area over this time period. However, the drop in forest area in the two aforementioned regions was so large, the percent forest area of the world decreased over this time period from __________________% to __________________%. 


## 3. COUNTRY-LEVEL DETAIL

### A. SUCCESS STORIES

There is one particularly bright spot in the data at the country level, ...

There is one particularly bright spot in the data at the country level, __________________. This country actually increased in forest area from 1990 to 2016 by __________________. It would be interesting to study what has changed in this country over this time to drive this figure in the data higher. The country with the next largest increase in forest area from 1990 to 2016 was the__________________, but it only saw an increase of __________________, much lower than the figure for __________________.

__________________ and __________________ are of course very large countries in total land area, so when we look at the largest percent change in forest area from 1990 to 2016, we aren’t surprised to find a much smaller country listed at the top. __________________ increased in forest area by __________________% from 1990 to 2016. 


### B. LARGEST CONCERNS

Which countries are seeing deforestation to the largest degree? We can answer this question in two ways. First, we can look at the absolute square kilometer decrease in forest area from 1990 to 2016. The following 3 countries had the largest decrease in forest area over the time period under consideration:

The second way to consider which countries are of concern is to analyze the data by percent decrease.

When we consider countries that decreased in forest area percentage the most between 1990 and 2016, we find that four of the top 5 countries on the list are in the region of __________________. The countries are __________________, __________________, __________________, and __________________. The 5th country on the list is __________________, which is in the __________________ region. 

From the above analysis, we see that __________________ is the only country that ranks in the top 5 both in terms of absolute square kilometer decrease in forest as well as percent decrease in forest area from 1990 to 2016. Therefore, this country has a significant opportunity ahead to stop the decline and hopefully spearhead remedial efforts.


### C. QUARTILES

The largest number of countries in 2016 were found in the __________________ quartile.

There were __________________ countries in the top quartile in 2016. These are countries with a very high percentage of their land area designated as forest. The following is a list of countries and their respective forest land, denoted as a percentage.


## 4. RECOMMENDATIONS